server/test/unit/server/content/test_storage.py

```diff
@@ -62,45 +62,53 @@ def test_exit(self):
 
 class TestFileStorage(TestCase):
 
-    @patch('pulp.server.content.storage.shutil')
     @patch('pulp.server.content.storage.config')
-    @patch('os.path.isdir', Mock(return_value=True))
-    def test_put_dir(self, config, shutil):
-        path_in = '/tmp/test/'
+    def test_get_path(self, config):
         storage_dir = '/tmp/storage'
-        unit = Mock(id='0123456789', _content_type_id='ABC')
         config.get = lambda s, p: {'server': {'storage_dir': storage_dir}}[s][p]
+        unit = Mock(id='0123456789', type_id='ABC')
+
+        # test
+        path = FileStorage.get_path(unit)
+
+        # validation
+        self.assertEqual(
+            path,
+            os.path.join(os.path.join(storage_dir, 'content', 'units'),
```

Comment:
Out of curiosity, what's the reason for the nested call to ``join``? 
Is ``os.path.join(storage_dir, 'content', 'units', unit.type_id, unit.id[0:4], unit.id)`` not equivalent?